packages/node-core/src/indexer/benchmark.service.ts

```diff
@@ -27,53 +27,120 @@ export class BenchmarkService {
   private currentProcessedBlockAmount?: number;
   private lastProcessedBlockAmount?: number;
 
+  private lastPoiWithMmr?: number;
+  private lastRegisteredPoiWithMmr?: number;
+  private poiTarget?: number;
+  private lastProcessedPoiAmount?: number;
+  private currentProcessedPoiAmount?: number;
+  private poiPerSecond?: number;
+  private currentPoiTimestamp?: number;
+  private lastRegisteredPoiTimestamp?: number;
+  // If add more benchmarking, we can make this class as a baseClass
+
   constructor(private nodeConfig: NodeConfig) {}
 
-  @Interval(SAMPLING_TIME_VARIANCE * 1000)
-  async benchmark(): Promise<void> {
-    try {
-      if (!this.currentProcessingHeight || !this.currentProcessingTimestamp) {
+  private async benchMarkPoi(): Promise<void> {
```

Comment:
It looks like we could make a generic version of this for POI and indexing to avoid duplicating code.